File: examples/nlp_examples.js
/**
 * Brainiall NLP API — JavaScript Examples
 *
 * Base URL: https://apim-ai-apis.azure-api.net/v1/nlp
 * Get your API key at https://brainiall.com
 */

const BASE_URL = "https://apim-ai-apis.azure-api.net/v1/nlp";
const API_KEY = "YOUR_KEY"; // Replace with your API key

async function nlpRequest(endpoint, text) {
  const response = await fetch(`${BASE_URL}/${endpoint}`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${API_KEY}`,
    },
    body: JSON.stringify({ text }),
  });
  return response.json();
}

// Toxicity Detection
async function toxicityExample() {
  console.log("=== Toxicity Detection ===");
  const texts = [
    "You are a wonderful person!",
    "I hate everything about this",
    "The weather is nice today",
  ];

  for (const text of texts) {
    const result = await nlpRequest("toxicity", text);
    const status = result.is_toxic ? "TOXIC" : "SAFE";
    console.log(`[${status}] ${text}`);
    console.log(
      `  Max: ${result.max_category} = ${result.max_score.toFixed(4)}`
    );
  }
}

// Sentiment Analysis
async function sentimentExample() {
  console.log("\n=== Sentiment Analysis ===");
  const reviews = [
    "Absolutely love this product!",
    "Terrible quality, very disappointed.",
    "It's okay, nothing special.",
    "Best purchase I've ever made!",
    "Would not recommend to anyone.",
  ];

  for (const review of reviews) {
    const result = await nlpRequest("sentiment", review);
    const icon = result.sentiment === "positive" ? "+" : "-";
    console.log(`[${icon} ${result.confidence.toFixed(3)}] ${review}`);
  }
}

// Named Entity Recognition
async function nerExample() {
  console.log("\n=== Named Entity Recognition ===");
  const text =
    "Elon Musk announced that SpaceX will launch from Cape Canaveral next month.";

  const result = await nlpRequest("entities", text);
  console.log(`Text: ${text}`);
  console.log(`Entities found: ${result.entity_count}`);
  for (const entity of result.entities) {
    console.log(
      `  [${entity.label}] ${entity.text} (${entity.score.toFixed(4)})`
    );
  }
}

// PII Detection
async function piiExample() {
  console.log("\n=== PII Detection ===");
  const text =
    "Contact alice@company.com or call 555-867-5309. SSN: 078-05-1120.";

  const result = await nlpRequest("pii", text);
  console.log(`Text: ${text}`);
  console.log(`PII found: ${result.pii_found} (${result.pii_count} items)`);
  for (const entity of result.entities) {
    console.log(`  [${entity.type}] ${entity.value}`);
  }
}

// Language Detection
async function languageExample() {
  console.log("\n=== Language Detection ===");
  const texts = [
    "Hello, how are you?",
    "Bonjour, comment allez-vous?",
    "Hallo, wie geht es Ihnen?",
    "こんにちは",
    "Olá, tudo bem?",
  ];

  for (const text of texts) {
    const result = await nlpRequest("language", text);
    console.log(`[${result.language} ${result.confidence.toFixed(3)}] ${text}`);
  }
}

// Full Pipeline
async function fullPipeline() {
  console.log("\n=== Full Analysis Pipeline ===");
  const text =
    "John from Google in NYC emailed john@google.com — fantastic product!";

  const endpoints = ["toxicity", "sentiment", "entities", "pii", "language"];
  const results = {};

  await Promise.all(
    endpoints.map(async (ep) => {
      results[ep] = await nlpRequest(ep, text);
    })
  );

  console.log(`Text: ${text}`);
  console.log(`Language: ${results.language.language}`);
  console.log(
    `Sentiment: ${results.sentiment.sentiment} (${results.sentiment.confidence.toFixed(3)})`
  );
  console.log(`Toxic: ${results.toxicity.is_toxic}`);
  console.log(`Entities: ${results.entities.entity_count}`);
  console.log(`PII: ${results.pii.pii_count} items`);
}

// Content Moderation
async function contentModeration() {
  console.log("\n=== Content Moderation ===");
  const messages = [
    "Hello, great to meet you!",
    "Email me at user@email.com",
    "My SSN is 123-45-6789",
  ];

  for (const msg of messages) {
    const [tox, pii] = await Promise.all([
      nlpRequest("toxicity", msg),
      nlpRequest("pii", msg),
    ]);

    let action;
    if (tox.is_toxic) action = "BLOCK (toxic)";
    else if (pii.pii_found) action = `REDACT (${pii.pii_count} PII)`;
    else action = "ALLOW";

    console.log(`[${action.padEnd(20)}] ${msg}`);
  }
}

// Run all examples
await toxicityExample();
await sentimentExample();
await nerExample();
await piiExample();
await languageExample();
await fullPipeline();
await contentModeration();

File: examples/nlp_examples.py
"""
Brainiall NLP API — Python Examples

Base URL: https://apim-ai-apis.azure-api.net/v1/nlp
Get your API key at https://brainiall.com
"""

import requests
import json
from concurrent.futures import ThreadPoolExecutor

BASE_URL = "https://apim-ai-apis.azure-api.net/v1/nlp"
HEADERS = {
    "Authorization": "Bearer YOUR_KEY",  # Replace with your API key
    "Content-Type": "application/json"
}


def toxicity_detection():
    """Detect toxic content across 6 categories."""
    texts = [
        "You are such a wonderful person!",
        "I hate everything about this terrible product",
        "The weather is nice today"
    ]

    for text in texts:
        response = requests.post(
            f"{BASE_URL}/toxicity",
            headers=HEADERS,
            json={"text": text}
        )
        result = response.json()
        status = "TOXIC" if result["is_toxic"] else "SAFE"
        print(f"[{status}] {text}")
        print(f"  Max: {result['max_category']} = {result['max_score']:.4f}")


def sentiment_analysis():
    """Classify text sentiment as positive or negative."""
    reviews = [
        "Absolutely love this! Best purchase I've ever made.",
        "Terrible quality. Completely disappointed.",
        "It's okay, nothing special but gets the job done.",
        "Amazing customer support, they went above and beyond!",
        "Would not recommend to anyone. Total waste of money."
    ]

    for review in reviews:
        response = requests.post(
            f"{BASE_URL}/sentiment",
            headers=HEADERS,
            json={"text": review}
        )
        result = response.json()
        icon = "+" if result["sentiment"] == "positive" else "-"
        print(f"[{icon} {result['confidence']:.3f}] {review}")


def named_entity_recognition():
    """Extract named entities: persons, organizations, locations."""
    text = "Satya Nadella, CEO of Microsoft, met with Tim Cook from Apple at their office in Cupertino, California."

    response = requests.post(
        f"{BASE_URL}/entities",
        headers=HEADERS,
        json={"text": text}
    )
    result = response.json()

    print(f"Text: {text}")
    print(f"Entities found: {result['entity_count']}")
    for entity in result["entities"]:
        print(f"  [{entity['label']:4s}] {entity['text']} (confidence: {entity['score']:.4f})")


def pii_detection():
    """Detect personally identifiable information."""
    text = "Please send the invoice to alice@company.com or call her at 555-867-5309. Her SSN is 078-05-1120."

    response = requests.post(
        f"{BASE_URL}/pii",
        headers=HEADERS,
        json={"text": text}
    )
    result = response.json()

    print(f"Text: {text}")
    print(f"PII found: {result['pii_found']} ({result['pii_count']} items)")
    for entity in result["entities"]:
        print(f"  [{entity['type']:11s}] {entity['value']}")


def language_detection():
    """Detect language from 217 supported languages."""
    texts = [
        "Hello, how are you doing today?",
        "Bonjour, comment allez-vous aujourd'hui?",
        "Hallo, wie geht es Ihnen heute?",
        "Hola, como estas hoy?",
        "Ciao, come stai oggi?",
        "Olá, como você está hoje?",
        "Привет, как дела сегодня?",
        "こんにちは、今日はお元気ですか？",
        "안녕하세요, 오늘 어떠세요?",
        "مرحبا، كيف حالك اليوم؟"
    ]

    for text in texts:
        response = requests.post(
            f"{BASE_URL}/language",
            headers=HEADERS,
            json={"text": text}
        )
        result = response.json()
        print(f"[{result['language']:3s} {result['confidence']:.3f}] {text[:50]}")


def full_analysis_pipeline():
    """Run all NLP analyses on a single text."""
    text = "John Smith from Amazon in Seattle emailed john@amazon.com — this new AI product is incredible!"

    endpoints = ["toxicity", "sentiment", "entities", "pii", "language"]
    results = {}

    for endpoint in endpoints:
        response = requests.post(
            f"{BASE_URL}/{endpoint}",
            headers=HEADERS,
            json={"text": text}
        )
        results[endpoint] = response.json()

    print(f"=== Full Analysis ===")
    print(f"Text: {text}")
    print(f"Language: {results['language']['language']} ({results['language']['confidence']:.4f})")
    print(f"Sentiment: {results['sentiment']['sentiment']} ({results['sentiment']['confidence']:.4f})")
    print(f"Toxic: {results['toxicity']['is_toxic']} (max: {results['toxicity']['max_score']:.4f})")
    print(f"Entities ({results['entities']['entity_count']}):")
    for e in results['entities']['entities']:
        print(f"  [{e['label']}] {e['text']}")
    print(f"PII ({results['pii']['pii_count']}):")
    for p in results['pii']['entities']:
        print(f"  [{p['type']}] {p['value']}")


def batch_processing():
    """Process multiple texts in parallel."""
    texts = [f"Review {i}: {'Great' if i % 2 == 0 else 'Terrible'} product!" for i in range(20)]

    def analyze(text):
        resp = requests.post(f"{BASE_URL}/sentiment", headers=HEADERS, json={"text": text})
        return resp.json()

    import time
    start = time.time()
    with ThreadPoolExecutor(max_workers=10) as pool:
        results = list(pool.map(analyze, texts))
    elapsed = time.time() - start

    positive = sum(1 for r in results if r["sentiment"] == "positive")
    print(f"Processed {len(texts)} texts in {elapsed:.2f}s")
    print(f"Positive: {positive}, Negative: {len(texts) - positive}")


def content_moderation():
    """Content moderation pipeline combining toxicity + PII checks."""
    messages = [
        "Great to meet you!",
        "Contact me at user@email.com",
        "I absolutely hate this stupid thing",
        "My SSN is 123-45-6789 and card is 4111-1111-1111-1111",
    ]

    for msg in messages:
        tox = requests.post(f"{BASE_URL}/toxicity", headers=HEADERS, json={"text": msg}).json()
        pii = requests.post(f"{BASE_URL}/pii", headers=HEADERS, json={"text": msg}).json()

        if tox["is_toxic"]:
            action = "BLOCK (toxic)"
        elif pii["pii_found"]:
            action = f"REDACT ({pii['pii_count']} PII)"
        else:
            action = "ALLOW"

        print(f"[{action:25s}] {msg}")


if __name__ == "__main__":
    print("=== Toxicity Detection ===")
    toxicity_detection()

    print("\n=== Sentiment Analysis ===")
    sentiment_analysis()

    print("\n=== Named Entity Recognition ===")
    named_entity_recognition()

    print("\n=== PII Detection ===")
    pii_detection()

    print("\n=== Language Detection ===")
    language_detection()

    print("\n=== Full Pipeline ===")
    full_analysis_pipeline()

    print("\n=== Batch Processing ===")
    batch_processing()

    print("\n=== Content Moderation ===")
    content_moderation()
